pkg/dockerregistry/server/auth.go

```diff
@@ -196,7 +197,7 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				}
 				verifiedPrune = true
 			default:
-				if err := verifyImageStreamAccess(ctx, imageStreamNS, imageStreamName, verb, osClient); err != nil {
+				if err := authorizeRepositoryAccess(ctx, imageStreamNS, imageStreamName, verb, osClient); err != nil {
```

Comment:
I'd rather see this loop be aware of access records being skipped because they appear to be cross-repository mounts... if we skip an access record for that reason, this loop should ensure it is paired with an authorized push access record check